tests/importers/test_autoconfig.py

```diff
@@ -0,0 +1,123 @@
+import shutil
+from pathlib import Path
+from typing import List, Text, Set
+from unittest.mock import Mock
+
+import pytest
+from _pytest.capture import CaptureFixture
+from _pytest.monkeypatch import MonkeyPatch
+
+from rasa.constants import CONFIG_AUTOCONFIGURABLE_KEYS
+from rasa.importers import autoconfig
+from rasa.utils import io as io_utils
+
+CONFIG_FOLDER = "data/test_config"
+
+SOME_CONFIG = CONFIG_FOLDER + "/arbitrary_example_config.yml"
+DEFAULT_CONFIG = "rasa/importers/default_config.yml"
+
+
+@pytest.mark.parametrize(
+    "config_path, autoconfig_keys",
+    [
+        ("rasa/cli/initial_project/config.yml", ["pipeline", "policies"]),
+        (CONFIG_FOLDER + "/config_pipeline_only.yml", ["policies"]),
+        (CONFIG_FOLDER + "/config_policies_only.yml", ["pipeline"]),
+        (SOME_CONFIG, []),
+    ],
+)
+def test_get_auto_configuration(
+    config_path: Text, autoconfig_keys: Set, monkeypatch: MonkeyPatch
+):
+    monkeypatch.setattr(autoconfig, "dump_config", Mock())
+    actual = autoconfig.get_auto_configuration(config_path)
+
+    default = io_utils.read_config_file(DEFAULT_CONFIG)
+
+    expected = io_utils.read_config_file(config_path)
+    for key in autoconfig_keys:
+        expected[key] = default[key]
```

Comment:
that's covered by line 41, isn't it?

Reply:
Here we are setting `expected` to be the expected output of the auto configuration (= the input to the auto configuration plus previously unspecified keys filled with the defaults). In line 41 we're checking if the actual output of the auto config matches this.